ecl/resource_ecl_provider_connectivity_tenant_connection_request_v2.go

```diff
@@ -0,0 +1,274 @@
+		updateOpts.NameOther = d.Get("name_other").(string)
+	}
+
+	if d.HasChange("description_other") {
+		hasChange = true
+		updateOpts.DescriptionOther = d.Get("description_other").(string)
+	}
+
+	if d.HasChange("tags_other") {
+		hasChange = true
+		updateOpts.TagsOther = getTags(d, "tags_other")
+	}
+
+	if hasChange {
+		r := tenant_connection_requests.Update(client, d.Id(), updateOpts)
+		if r.Err != nil {
+			return fmt.Errorf("error updating ECL Provider Connectivity Tenant Connection Request: %s", r.Err)
+		}
+		log.Printf("[DEBUG] Tenant Connection Request has successfully updated.")
+	}
+
+	return resourceProviderConnectivityTenantConnectionRequestV2Read(d, meta)
+}
+
+func resourceProviderConnectivityTenantConnectionRequestV2Delete(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+	client, err := config.providerConnectivityV2Client(GetRegion(d, config))
+	if err != nil {
+		return fmt.Errorf("error creating ECL Provider Connectivity client: %s", err)
+	}
+
+	if err := tenant_connection_requests.Delete(client, d.Id()).ExtractErr(); err != nil {
+		return fmt.Errorf("error deleting ECL Provider Connectivity Tenant Connection Request: %s", err)
+	}
+
+	stateConf := &resource.StateChangeConf{
+		Target:     []string{"DELETED"},
+		Refresh:    waitForTenantConnectionRequestStateDelete(client, d.Id()),
+		Timeout:    d.Timeout(schema.TimeoutDelete),
+		Delay:      5 * time.Second,
+		MinTimeout: 3 * time.Second,
+	}
+
+	_, err = stateConf.WaitForState()
+	if err != nil {
+		return fmt.Errorf("Error deleting ECL tenant connection request (%s): %s", d.Id(), err)
+	}
+
+	d.SetId("")
+
+	return nil
+}
+
+func getTags(d *schema.ResourceData, tagName string) map[string]string {
+	rawTags := d.Get(tagName).(map[string]interface{})
+	tags := map[string]string{}
+	for key, value := range rawTags {
+		if v, ok := value.(string); ok {
+			tags[key] = v
+		}
+	}
+	return tags
+}
+
+func waitForTenantConnectionRequestCreate(connClient *eclcloud.ServiceClient, sssClient *eclcloud.ServiceClient, id string) resource.StateRefreshFunc {
+	return func() (interface{}, string, error) {
+		request, err := tenant_connection_requests.Get(connClient, id).Extract()
+
+		if err == nil && approvalRequestExists(sssClient, request.ApprovalRequestID) {
+			return request, request.Status, nil
+		}
+
+		return nil, "", err
+	}
+}
+
+func approvalRequestExists(sssClient *eclcloud.ServiceClient, approvalRequestID string) bool {
+	_, err := approval_requests.Get(sssClient, approvalRequestID).Extract()
+	if err != nil {
+		return false
+	}
+	return true
+}
+
+func waitForTenantConnectionRequestStateDelete(client *eclcloud.ServiceClient, id string) resource.StateRefreshFunc {
+	return func() (interface{}, string, error) {
+		log.Printf("[DEBUG] Attempting to delete ECL tenant connection request %s.\n", id)
+		request, err := tenant_connection_requests.Get(client, id).Extract()
+		if err != nil {
+			if _, ok := err.(eclcloud.ErrDefault404); ok {
```

Comment:
From Go 1.13, spec of error handling has been changed. Other parts of our code has not been updated yet...
https://speakerdeck.com/ciarana/lets-talk-about-error-handling-after-go-1-13?slide=5
```suggestion
			var e *eclcloud.ErrDefault404
			if errors.As(err, &e) {
```

Reply:
I`ll do it, thanks for the great information!